airbyte-integrations/bases/base-typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/AirbyteTypeUtils.java

```diff
@@ -68,12 +68,13 @@ private static boolean nodeIsOrContainsType(final JsonNode node, final String ty
   }
 
   protected static AirbyteType getAirbyteProtocolType(final JsonNode node) {
+    if (node.isTextual()) {
+      return AirbyteProtocolType.matches(node.asText());
+    }
+
     final JsonNode propertyType = node.get("type");
     final JsonNode airbyteType = node.get("airbyte_type");
     final JsonNode format = node.get("format");
-    if (Stream.of(propertyType, airbyteType, format).allMatch(Objects::isNull)) {
-      return AirbyteProtocolType.matches(node.asText());
```

Comment:
We could also return `AirbyteProtocolType.UNKNOWN` here but this case is correctly taken care of by the rest of the code (i.e., `nodeIsOrContainsType` handles null nodes already so all the conditions will be skipped).